x-pack/plugins/canvas/public/components/arg_form/index.js

```diff
@@ -5,29 +5,44 @@
  * 2.0.
  */
 
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { compose, withState, lifecycle } from 'recompose';
 import { getAssets } from '../../state/selectors/assets';
 import { getWorkpadInfo } from '../../state/selectors/workpad';
 import { ArgForm as Component } from './arg_form';
 
-export const ArgForm = compose(
-  withState('label', 'setLabel', ({ label, argTypeInstance }) => {
-    return label || argTypeInstance.displayName || argTypeInstance.name;
-  }),
-  withState('resolvedArgValue', 'setResolvedArgValue'),
-  withState('renderError', 'setRenderError', false),
-  lifecycle({
-    componentDidUpdate(prevProps) {
-      if (prevProps.templateProps.argValue !== this.props.templateProps.argValue) {
-        this.props.setRenderError(false);
-        this.props.setResolvedArgValue();
-      }
-    },
-  }),
-  connect((state) => ({ workpad: getWorkpadInfo(state), assets: getAssets(state) }))
-)(Component);
+const getLabel = (label, argTypeInstance) =>
+  label || argTypeInstance.displayName || argTypeInstance.name;
+
+const ArgFormComponent = (props) => {
+  const { argTypeInstance, label: labelFromProps, templateProps } = props;
+  const [label, setLabel] = useState(getLabel(labelFromProps, argTypeInstance));
+  const [resolvedArgValue, setResolvedArgValue] = useState(null);
+  const [renderError, setRenderError] = useState(false);
+
+  useEffect(() => {
+    setRenderError(false);
+    setResolvedArgValue();
+  }, [templateProps?.argValue]);
+
+  return (
+    <Component
+      {...props}
+      label={label}
+      setLabel={setLabel}
+      resolvedArgValue={resolvedArgValue}
+      setResolvedArgValue={setResolvedArgValue}
+      renderError={renderError}
+      setRenderError={setRenderError}
+    />
+  );
+};
+
+export const ArgForm = connect((state) => ({
```

Comment:
While you're in here, can you swap this to use `useSelector` hook instead of connect.

Reply:
Done.